src/main/java/com/jfb/tickets/service/UserService.java

```diff
@@ -24,10 +35,20 @@ public List<User> getUsers() {
     }
 
     public void saveUser(User user) {
+        System.out.println("Transaction open? : " + TransactionSynchronizationManager.isActualTransactionActive());
         userDAO.save(user);
     }
 
     public void deleteUser(UUID userId) {
         userDAO.delete(userId);
     }
+
+    public void updateUserStatusAndCreateTicket(UUID userId, boolean status, Ticket ticket) {
+        if(!allowUserUpdateAndTicketCreate) {
+            throw new RuntimeException("User status update and ticket creation are not allowed.");
+        }
+        System.out.println("Transaction open? : " + TransactionSynchronizationManager.isActualTransactionActive());
+        userDAO.updateUserStatus(userId, status);
+        ticketDAO.save(ticket, userId);
+    }
```

Comment:
this method will not be performed under 1 transaction, and each task  `userDAO.updateUserStatus(...)` and `ticketDAO.save(...)`  will be done separately, because UserDAO and TicketDAO classes work with PreparedStatement object, which is not controlled by Spring.

Spring manipulates with JdbcTemplate object. You need to:
1. Create bean of `JdbcTemplate` in your config class
2. Inject this bean in DAO classes
3. make a call to db on this object, for example:
```java
    public void updateUserStatus(UUID userId, boolean status) {
        jdbcTemplate.update(UPDATE_USER_STATUS_QUERY, status, userId);
    }
```

If both methods `userDAO.updateUserStatus(...)` and `ticketDAO.save(...)` will use  `jdbcTemplate` inside, then this process will be controlled by Spring transaction and, in fail case, rolled back.